File: javaScript/problem3.js
// Find the largest prime factor of nth number

function largestPrimefactor(num) {
  var factor = 2;

  while (num > factor) {
    if (num % factor === 0) {
      num = num / factor;
    } else {
      factor++;
    }
  }
  return factor;
}

console.log(largestPrimefactor(600851475143));

File: rubyChallenge/problem3.rb
# find the largest prime factor of nth number

def find_largest_prime_factor num
  factor = 2

  while num > factor
    if num % factor == 0
      num = num / factor
    else
      factor += 1
    end
  end
  factor
end

p find_largest_prime_factor 600851475143
